File: lib-database-entities/src/main/java/com/ustadmobile/lib/db/entities/SaleListDetail.java
package com.ustadmobile.lib.db.entities;

/**
 * Sale 's POJO for representing itself on the view (and recycler views)
 */
public class SaleListDetail extends Sale {

    String locationName;
    float saleAmount;
    String saleCurrency;
    int saleItemCount;
    String saleTitleGen;
    float saleAmountPaid;
    float saleAmountDue;
    long earliestDueDate;


    public String getLocationName() {
        return locationName;
    }

    public void setLocationName(String locationName) {
        this.locationName = locationName;
    }

    public float getSaleAmount() {
        return saleAmount;
    }

    public void setSaleAmount(float saleAmount) {
        this.saleAmount = saleAmount;
    }

    public String getSaleCurrency() {
        return saleCurrency;
    }

    public void setSaleCurrency(String saleCurrency) {
        this.saleCurrency = saleCurrency;
    }

    public int getSaleItemCount() {
        return saleItemCount;
    }

    public void setSaleItemCount(int saleItemCount) {
        this.saleItemCount = saleItemCount;
    }

    public String getSaleTitleGen() {
        return saleTitleGen;
    }

    public void setSaleTitleGen(String saleTitleGen) {
        this.saleTitleGen = saleTitleGen;
    }

    public float getSaleAmountPaid() {
        return saleAmountPaid;
    }

    public void setSaleAmountPaid(float saleAmountPaid) {
        this.saleAmountPaid = saleAmountPaid;
    }

    public float getSaleAmountDue() {
        return saleAmountDue;
    }

    public void setSaleAmountDue(float saleAmountDue) {
        this.saleAmountDue = saleAmountDue;
    }

    public long getEarliestDueDate() {
        return earliestDueDate;
    }

    public void setEarliestDueDate(long earliestDueDate) {
        this.earliestDueDate = earliestDueDate;
    }
}

File: lib-database-entities/src/commonMain/kotlin/com/ustadmobile/lib/db/entities/SaleListDetail.kt
package com.ustadmobile.lib.db.entities

import kotlinx.serialization.Serializable
import androidx.room.Embedded


/**
 * Sale 's POJO for representing itself on the view (and recycler views)
 */
@Serializable
class SaleListDetail() : Sale() {

    @Embedded
    var customer: Person? = null

    //String categoryName;

    var locationName: String? = null
    var saleAmount: Float = 0.toFloat()
    var saleCurrency: String? = null
    var saleItemCount: Int = 0
    var saleTitleGen: String? = null
    var saleTitleGenDari: String? = null
    var saleTitleGenPashto: String? = null
    var saleProductNames: String? = null
    var saleProductNamesDari: String? = null
    var saleProductNamesPashto: String? = null
    var saleAmountPaid: Float = 0.toFloat()
    var saleAmountDue: Float = 0.toFloat()
    var earliestDueDate: Long = 0
    var saleItemPreOrder: Boolean = false
    var saleCreator: String? = null


}